### PR TITLE
Created the base of the lexer

### PR DESCRIPTION
**Pull request by adiman9**
The lexer is the part of the interpreter that takes a source file as a string as input and turns it into a list of tokens that represents the syntax that occurs within the file.

This commit lays out the basis of the lexer. It can accept a string as input and has a tokenise method that loops through all the characters and counts the lines in the file. It also has many other methods that allows it to traverse the source string. 

It is a solid base to start building from.

### CODE REVIEW

**Review by cpbadman — APPROVED** [review]
Looks good! I understand exactly what you've done for the most part, will wait for Zhanet to review before merging

**Comment by adiman9** [comment]
awesome. I was going to say you should wait for zhanet to review before merging but you beat me to it

**Comment by mushmula27** [comment]
Reviewed and understood.

**Review by mushmula27 — APPROVED** [review]

**Pull request merged**
